src/redisearch_rs/encode_decode/src/varint/mod.rs

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! This module provides functions to read and write varints (variable-length integers) in a
+//! compact format. Varints are used to encode integers in a way that uses fewer bytes for small
+//! values and more bytes for larger values. This is useful for reducing the size of data when
+//! transmitting or storing integers.
+//!
+//! The module includes functions to read and write varints from and to a byte buffer, as well as
+//! functions to read and write field masks, which are a specific type of varint used in the
+//! Redisearch project. The module also includes a [`VectorWriter`] struct that allows for writing
+//! multiple varints to a vector in a more efficient way.
+
+mod vector_writer;
+pub use vector_writer::VectorWriter;
+
+use super::FieldMask;
+
+use std::{
+    io::{self, Read, Write},
+    ops::{Deref, DerefMut},
+};
+
+/// Read an encoded integer from the given reader.
+pub fn read<R>(mut read: R) -> io::Result<u32>
+where
+    R: Read,
+{
+    let mut buffer = [0; 1];
+    read.read_exact(&mut buffer)?;
+    let mut c = buffer[0];
+
+    let mut val = (c & 127) as u32;
```

Comment:
Can we make the `127` a byte constant so that it is easier to see this is the value `0111_1111` and therefore easier to see the effect of this AND operation

Reply:
I'd rather keep it as if for now, so it's easy to see that the Rust impl is identical to the C one.

Reply:
I thinks it makes it easier for the reviewer (especially folks not too familiar with Rust) to easily compare. I'm not objecting to your suggestion but I think it'd be best to do after this PR.

Reply:
> I really don't see how anyone (a Rust or C person) would find `127` easier than a `0111_1111` for a binary operation

I didn't say they would. :)

Reply:
Let's leave this for a future improvement. I already made another note for a potential improvement too